src/Plugin/media/Source/MediaSourceBase.php

```diff
@@ -201,11 +201,23 @@ public function getAccount(): Account {
    *   The mpx object.
    */
   public function getMpxObject(MediaInterface $media): ObjectInterface {
+    static $cache = [];
+
+    if (isset($cache[$media->id()])) {
+      return $cache[$media->id()];
+    }
+
+    // To avoid memory issues, reset the cache once it contains 100 elements.
+    if (count($cache) == 100) {
+      $cache = [];
+    }
+
     $id = $media->get($this->configuration['source_field'])->getString();
     $factory = $this->dataObjectFactoryCreator->fromMediaSource($this);
 
-    return $factory->load(new Uri($id))->wait();
+    $cache[$media->id()] = $factory->load(new Uri($id))->wait();
 
+    return $cache[$media->id()];
```

Comment:
This is always going to return `TRUE`, because `id()` on a new entity returns null, which gets cast to an empty string. This leads to broken data on imports as all new media will have the same values as the first media item imported.

I'll work on fixing this.

Reply:
Whoa! Great catch @deviantintegral.